spark-3.0/src/main/scala/org/neo4j/spark/reader/SimpleScanBuilder.scala

```diff
@@ -0,0 +1,40 @@
+package org.neo4j.spark.reader
+
+import org.apache.spark.sql.connector.read.{Scan, ScanBuilder, SupportsPushDownFilters, SupportsPushDownRequiredColumns}
+import org.apache.spark.sql.sources.Filter
+import org.apache.spark.sql.types.StructType
+import org.neo4j.spark.util.{Neo4jOptions}
+
+class SimpleScanBuilder(neo4jOptions: Neo4jOptions, jobId: String, schema: StructType) extends ScanBuilder
+  with SupportsPushDownFilters
+  with SupportsPushDownRequiredColumns {
+
+  private var filters: Array[Filter] = Array[Filter]()
+
+  private var requiredColumns: StructType = new StructType()
+
+  override def build(): Scan = {
+    new SimpleScan(neo4jOptions, jobId, schema, filters, requiredColumns)
+  }
+
+  override def pushFilters(filtersArray: Array[Filter]): Array[Filter] = {
+    if (neo4jOptions.pushdownFiltersEnabled) {
+      filters = filtersArray
+    }
+
+    filtersArray
+  }
+
+  override def pushedFilters(): Array[Filter] = filters
+
+  override def pruneColumns(requiredSchema: StructType): Unit = {
+    requiredColumns = if (
```

Comment:
why this `if`?

Reply:
this controls weather we enable or disable PushDownColumnsSupport:

* if is disabled in the options
* if there is `nodeMap` enabled for relationships
* if the required schema is equals to the dataset schema